iphone/Maps/UI/PlacePage/Components/OpenInAppActionSheet/OpenInApplication.swift

```diff
@@ -0,0 +1,123 @@
+enum OpenInApplication: Int, CaseIterable {
+  case osm
+  case googleMaps
+  case appleMaps
+  case osmAnd
+  case yandexMaps
+  case dGis
+  case cityMapper
+  case moovit
+  case uber
+  case waze
+}
+
+extension OpenInApplication {
+
+  static var allCases: [OpenInApplication] {
+    [
+      .osm,
+      .googleMaps,
+      .appleMaps,
+      .osmAnd,
+      .yandexMaps,
+      .dGis,
+      .cityMapper,
+      .moovit, 
+      .uber,
+      .waze
+    ]
+  }
+
+  static var availableApps: [OpenInApplication] {
+    // OSM should always be first in the list.
+    let sortedApps: [OpenInApplication] = [.osm] + allCases.filter { $0 != .osm }.sorted(by: { $0.name < $1.name })
+    return sortedApps.filter { UIApplication.shared.canOpenURL(URL(string: $0.scheme)!) }
+  }
+
+  var name: String {
+    switch self {
+    case .osm:
+      return "OpenStreetMap"
+    case .googleMaps:
+      return "Google Maps"
+    case .appleMaps:
+      return "Apple Maps"
+    case .osmAnd:
+      return "OsmAnd Maps"
+    case .yandexMaps:
+      return "Yandex Maps"
+    case .dGis:
+      return "2GIS"
+    case .cityMapper:
+      return "Citymapper"
+    case .moovit:
+      return "Moovit"
+    case .uber:
+      return "Uber"
+    case .waze:
+      return "Waze"
+    }
+  }
+
+  // Schemes should be registered in LSApplicationQueriesSchemes - see Info.plist.
+  var scheme: String {
+    switch self {
+    case .osm:
+      return "https://osm.org/go/"
+    case .googleMaps:
+      return "comgooglemaps://"
+    case .appleMaps:
+      return "https://maps.apple.com/"
+    case .osmAnd:
+      return "osmandmaps://"
+    case .yandexMaps:
+      return "yandexmaps://"
+    case .dGis:
+      return "dgis://"
+    case .cityMapper:
+      return "citymapper://"
+    case .moovit:
+      return "moovit://"
+    case .uber:
+      return "uber://"
+    case .waze:
+      return "waze://"
+    }
+  }
+
+  func linkWith(coordinates: CLLocationCoordinate2D, zoomLevel: Int = Int(FrameworkHelper.currentZoomLevel()), destinationName: String? = nil) -> String {
```

Comment:
It will increase the overall code amount and implementation time but may save some time for the android because:
- it needs to implement all the logic into the c++ 
- objc-c wrapper for c++ (it will be great to use swift/c++ interop one day)
- jni wrappers for c++

Reply:
Ok, let's raise this question later when implementing a similar functionality for Android. It would be great to create a new issue (or modify the existing one) to leave it open for the missing Android implementation.